### PR TITLE
update build scripts

### PR DESCRIPTION
**Pull request by bavardage**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,47 +1,75 @@
...
     repositories {
         jcenter()
+        maven { url "http://palantir.bintray.com/releases" }
     }
 
     dependencies {
-        classpath "com.palantir:gradle-baseline-java:${baselineVersion}"
-        classpath "com.jfrog.bintray.gradle:gradle-bintray-plugin:${bintrayPlugin}"
-        classpath "org.jfrog.buildinfo:build-info-extractor-gradle:4.0.0"
+        classpath 'com.google.guava:guava:19.0'
```
Comment (markelliot) [inline build.gradle]: don't think we should have this

**Comment by bavardage** [comment]
ah, baseline is now java8... exciting....

**Comment by bavardage** [comment]
@markelliot thoughts on ECV continuing to support java7?

**Comment by markelliot** [comment]
I think we're good to drop Java 7 support at this point.

File: build.gradle [hunk 2]
```diff
@@ -1,54 +1,80 @@
...
+}
+
+apply plugin: 'com.palantir.baseline-config'
 apply plugin: 'com.palantir.baseline-idea'
+apply plugin: 'org.inferred.processors'
 
-repositories {
-    jcenter() // needed for baselineUpdateConfig task
+dependencies {
+    baseline group: 'com.palantir.baseline', name: 'gradle-baseline-java-config', version: '0.13.0-rc3', ext: 'zip'
```
Comment (markelliot) [inline build.gradle]: can omit the version w/ dependency recommender + versions.props

File: gradle/wrapper/gradle-wrapper.properties [hunk 3]
```diff
@@ -2,4 +2,4 @@ distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-2.14-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-3.4.1-bin.zip
```
Comment (markelliot) [inline gradle/wrapper/gradle-wrapper.properties:5]: go all the way to 3.5 now?

Comment (bavardage) [reply gradle/wrapper/gradle-wrapper.properties:5]: had some conflicts with errorprone on some other projects - will give it a whirl

Comment (bavardage) [reply gradle/wrapper/gradle-wrapper.properties:5]:
```
:encrypted-config-value:compileJava

FAILURE: Build failed with an exception.

* What went wrong:
net.ltgt.gradle.errorprone.ErrorProneToolChain.getVersion()Ljava/lang/String;
```

Comment (bavardage) [reply gradle/wrapper/gradle-wrapper.properties:5]: https://github.com/palantir/gradle-baseline/pull/158

File: build.gradle (hunk 2 above)
Comment (bavardage) [reply build.gradle]: prefer to leave buildscript dependency versions only in build.gradle, so you don't have to update buildscript deps across two files at least

Comment (markelliot) [reply build.gradle]: makes sense, then would use shorthand here as well

**Pull request merged**